VisualPinball.Unity/VisualPinball.Unity/VPT/Trough/TroughApi.cs

```diff
@@ -196,10 +197,14 @@ void IApi.OnInit(BallManager ballManager)
 
 			} else {
 				_stackSwitches = new DeviceSwitch[MainComponent.SwitchCount];
-				foreach (var sw in MainComponent.AvailableSwitches) {
-					if (sw.InternalId > 0) {
-						_stackSwitches[sw.InternalId - 1] = CreateSwitch(sw.Id, false, MainComponent.Type == TroughType.ModernOpto ? SwitchDefault.NormallyClosed : SwitchDefault.NormallyOpen);
-						_switchLookup[sw.Id] = _stackSwitches[sw.InternalId - 1];
+				foreach (var @switch in MainComponent.AvailableSwitches) {
+					var match = new Regex(@"^ball_switch_(\d+)$").Match(@switch.Id);
+					if (match.Success) {
+						int.TryParse(match.Groups[1].Value, out int id);
+						if (id > 0) {
+							_stackSwitches[id - 1] = CreateSwitch(@switch.Id, false, MainComponent.Type == TroughType.ModernOpto ? SwitchDefault.NormallyClosed : SwitchDefault.NormallyOpen);
+							_switchLookup[@switch.Id] = _stackSwitches[id - 1];
+						}

```

Comment:
This is a bit ugly, but since it's only done once I guess there isn't too much of an impact. I would at least move the regex compilation out of the loop though.

And one thing I would add is a comment when `ball_switch_{i}` is defined, so it's clear we cannot rename it without changing the regex.

Reply:
Done. Just curious, if there are multiple troughs on a playfield, would multiple `ball_switch_1`s  cause any issues?